include/qengine_opencl.hpp

```diff
@@ -74,25 +75,55 @@ class QEngineOCL : public QEngineCPU {
         maxQPower = 1 << qubitCount;
     }
 
+    virtual void EnableNormalize(bool doN) { doNormalize = doN; }
+    virtual real1 GetNorm(bool update = true)
+    {
+        if (update) {
+            UpdateRunningNorm();
+        }
+        return runningNorm;
+    }
+    virtual void SetNorm(real1 n) { runningNorm = n; }
+
+    // CL_MAP_READ = (1 << 0); CL_MAP_WRITE = (1 << 1);
+    virtual void LockSync(cl_int flags = 3);
```

Comment:
Out of curiosity, are there actual defined flags CL_MAP_READ etc that you can use instead of 3?  e.g. `flags = CL_MAP_READ | CL_MAP_WRITE`?

Reply:
I thought it wouldn't be able able to turn `(CL_MAP_READ | CL_MAP_WRITE)` into a compile-time constant default, but it turns out that it can. Done.